x-pack/solutions/observability/plugins/streams/server/lib/streams/client.ts

```diff
@@ -530,6 +572,22 @@ export class StreamsClient {
     return this.dependencies.scopedClusterClient.asCurrentUser.indices
       .getDataStream({ name })
       .then((response) => {
+        if (response.data_streams.length === 0) {
```

Comment:
This is where the original error came from, if there is no data stream this API does **not** throw any error, it simply returns an empty list.

Which caused the flow to continue to try to translate this undefined object into a Classic stream definition hence causing the `undefined.name` error.